lucetc/src/decls.rs

```diff
@@ -72,15 +72,17 @@ pub struct ModuleDecls<'a> {
 
 impl<'a> ModuleDecls<'a> {
     pub fn new<B: ClifBackend>(
-        info: ModuleInfo<'a>,
+        mut info: ModuleInfo<'a>,
         clif_module: &mut ClifModule<B>,
         bindings: &Bindings,
         runtime: Runtime,
         heap_settings: HeapSettings,
     ) -> Result<Self, LucetcError> {
-        let (function_names, imports, exports) = Self::declare_funcs(&info, clif_module, bindings)?;
+        let (mut function_names, imports, exports) =
+            Self::declare_funcs(&info, clif_module, bindings)?;
         let table_names = Self::declare_tables(&info, clif_module)?;
-        let runtime_names = Self::declare_runtime(&runtime, clif_module)?;
+        let runtime_names =
+            Self::declare_runtime(&mut info, &mut function_names, &runtime, clif_module)?;
```

Comment:
Because we're tracking more information in `ModuleData` (built out of this and `ModuleInfo`), the amount of bookkeeping as functions are introduced has grown a bit. I'm thinking about factoring out most of it into one or two functions on `ModuleDecls` to set up signatures, names, and indices. Then that can be reused by the typical `ModuleEnvironment` path of function declaration, or for internal declarations like runtime functions or the stack probe.

Reply:
did this